atomic_reactor/plugins/post_gather_builds_metadata.py

```diff
@@ -0,0 +1,136 @@
+"""
+Copyright (c) 2017, 2019 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+import os
+from typing import Optional, Any, Dict
+
+from atomic_reactor.config import get_koji_session
+from atomic_reactor.plugin import PostBuildPlugin
+from atomic_reactor.constants import PLUGIN_GATHER_BUILDS_METADATA_KEY
+from atomic_reactor.util import is_scratch_build, map_to_user_params, get_platforms
+from atomic_reactor.utils.koji import get_buildroot, get_output, KojiUploadLogger
+from osbs.utils import ImageName
+
+
+class GatherBuildsMetadataPlugin(PostBuildPlugin):
+    """
+    Gather builds metadata from platform-specific builds which are done by
+    podman-remote on remote hosts. This metadata may also contain metadata
+    fetched from remote hosts by running commands via ssh tunnel.
+
+    This plugin returns a mapping from platform to the build's metadata.
+    For example,
+
+    {
+        "x86_64": {...},
+        "s390x": {...},
+    }
+
+    Each metadata mapping follows the format of Koji Content Generator Metadata.
+    This plugin ensures it has keys ``metadata_version``, ``buildroots`` and
+    ``output``.
+    """
+
+    key = PLUGIN_GATHER_BUILDS_METADATA_KEY
+    is_allowed_to_fail = False
+
+    args_from_user_params = map_to_user_params("koji_upload_dir")
```

Comment:
If I recall correctly, this user param used to be generated by the orchestrator plugin. We will not have it in user params anymore, but maybe we can just generate it here. I think another plugin uses the directory as well, we should make sure it's the same directory for both.

Reply:
Addressed in later commit, but this is still relevant for `KojiImportPlugin.get_server_dir()` - now it should generate the upload dir instead of getting it from the orchestrator plugin